src/models/stores/sorted-documents.ts

```diff
@@ -254,14 +348,14 @@ export class SortedDocuments {
       documentKey: metadataDoc?.key,
       type: metadataDoc?.type as any,
       title: metadataDoc?.title || undefined,
-      properties: metadataDoc?.properties,
+      properties: metadataDoc?.properties.toJSON(),
       userId: metadataDoc?.uid,
       groupId: undefined,
       visibility,
       originDoc: undefined,
       pubVersion: undefined,
-      problem: metadataDoc?.problem,
-      investigation: metadataDoc?.investigation,
+      problem: metadataDoc?.problem || undefined,
+      investigation: metadataDoc?.investigation || undefined,
```

Comment:
This confused me a little and I want to be sure I understand. Did you add ` || undefined` here because `metadataDoc.problem` and `metadataDoc.investigation` are typed as `maybeNull` and we should use `undefined` here instead if they are `null`?